File: src/main/java/gg/mineral/practice/commands/settings/TogglePlayerVisibilityCommand.java
package gg.mineral.practice.commands.settings;

import gg.mineral.practice.commands.PlayerCommand;

import gg.mineral.practice.managers.ProfileManager;
import gg.mineral.practice.util.messages.impl.ChatMessages;
import lombok.val;

public class TogglePlayerVisibilityCommand extends PlayerCommand {

	public TogglePlayerVisibilityCommand() {
		super("toggleplayervisibility");
	}

	@Override
	public void execute(org.bukkit.entity.Player pl, String[] args) {
		val profile = ProfileManager.getOrCreateProfile(pl);
		profile.setPlayersVisible(!profile.isPlayersVisible());

		profile.updateVisiblity();

		ChatMessages.VISIBILITY_TOGGLED.clone().replace("%toggled%",
				profile.isPlayersVisible() ? "enabled" : "disabled").send(pl);
	}
}

File: src/main/kotlin/gg/mineral/practice/commands/settings/TogglePlayerVisibilityCommand.kt
package gg.mineral.practice.commands.settings

import dev.rollczi.litecommands.annotations.command.Command
import dev.rollczi.litecommands.annotations.context.Context
import dev.rollczi.litecommands.annotations.execute.Execute
import gg.mineral.practice.entity.Profile
import gg.mineral.practice.util.messages.impl.ChatMessages

@Command(name = "toggleplayervisibility")
class TogglePlayerVisibilityCommand {
    @Execute
    fun execute(@Context profile: Profile) {
        profile.playersVisible = !profile.playersVisible

        profile.updateVisibility()

        profile.message(
            ChatMessages.VISIBILITY_TOGGLED.clone().replace(
                "%toggled%",
                if (profile.playersVisible) "enabled" else "disabled"
            )
        )
    }
}
